server.go

```diff
@@ -1035,206 +1038,233 @@ func newServer(listenAddrs []net.Addr, chanDB *channeldb.DB, cc *chainControl,
 // Started returns true if the server has been started, and false otherwise.
 // NOTE: This function is safe for concurrent access.
 func (s *server) Started() bool {
-	return atomic.LoadInt32(&s.started) != 0
+	return atomic.LoadInt32(&s.active) != 0
 }
 
 // Start starts the main daemon server, all requested listeners, and any helper
 // goroutines.
 // NOTE: This function is safe for concurrent access.
 func (s *server) Start() error {
-	// Already running?
-	if !atomic.CompareAndSwapInt32(&s.started, 0, 1) {
-		return nil
-	}
-
-	if s.torController != nil {
-		if err := s.initTorController(); err != nil {
-			return err
+	var startErr error
+	s.start.Do(func() {
```

Comment:
Hm, not sure if I prefer this new pattern... (indentation and complicates error handling)

What about 
```go
if !atomic.CompareAndSwapInt32(&s.started, 0, 1) {
     return nil
}
defer atomic.StoreInt32(&s.active, 1)
```

Reply:
We're moving to using `sync.Once` elsewhere, and if the indentation gets too out of hand, we can also lift it into a new method rather than using a function closure.